src/SpectralUnmixing.jl

```diff
@@ -45,42 +45,92 @@ export initiate_output_datasets, set_band_names, write_results
+  bad wavelength regions [1300, 1500] and [1800, 2000].
+  - A specific wavelength (as a string): The data will be normalized using the reflectance
+  value at this wavelength.
+
+# Returns
+- A matrix of scaled reflectance data with same dimensions as `refl`.
+"""
 function scale_data(refl::Matrix{Float64}, wavelengths::Vector{Float64}, criteria::String)
 
     if criteria == "none"
         return refl
     elseif criteria == "brightness"
-        bad_regions_wl = [[1300,1500],[1800,2000]]
+        bad_regions_wl = [[1300, 1500], [1800, 2000]]
         good_bands = convert(Vector{Bool}, ones(length(wavelengths)))
         for br in bad_regions_wl
             good_bands[wl_index(wavelengths, br[1]):wl_index(wavelengths, br[2])] .= false
         end
         if length(size(refl)) == 2
-            norm = sqrt.(mean(refl[:,good_bands].^2, dims=2))
+            norm = sqrt.(mean(refl[:, good_bands] .^ 2, dims=2))
         else
-            norm = sqrt.(mean(refl[good_bands].^2))
+            norm = sqrt.(mean(refl[good_bands] .^ 2))
         end
     else
         try
-            target_wl = parse(Float64,criteria)
+            target_wl = parse(Float64, criteria)
             if length(size(refl)) == 2
-                norm = refl[:,wl_index(wavelengths, target_wl)] ./ 0.5
+                norm = refl[:, wl_index(wavelengths, target_wl)] ./ 0.5
             else
                 norm = refl[wl_index(wavelengths, target_wl)] ./ 0.5
             end
         catch e
-            throw(ArgumentError(string("normalization must be [none, brightness, or a specific wavelength].  Provided:", criteria)))
+            throw(ArgumentError(string("normalization must be [none, brightness, or a
+            specific wavelength].  Provided:", criteria)))
         end
     end
 
     return refl ./ norm
 end
 
-function get_sma_permutation(class_idx, num_endmembers::Vector{Int64}, combination_type::String, library_length::Int64)
+"""
+    get_sma_permutation(class_idx, num_endmembers::Vector{Int64},
+                        combination_type::String, library_length::Int64)
+
+Generate a permutation of endmember indices for SMA.
+
+# Arguments
+- `class_idx`: A collection (such as a vector of vectors) of size n where each element
+is a set of indices corresponding to endmembers belonging to one class.
+- `num_endmembers::Vector{Int64}`: Desired number of endmembers to select. The first
+element specifies the number of endmembers to permute; if it is `-1`, all endmembers in
+library will be included.
+- `combination_type::String`: Permutation type options:
+  - `"class-even"`: Randomly select one endmember from each class until the required number
+  of endmembers is selected.
+  - Any other types: Randomly selects endmembers from the entire library.
+- `library_length::Int64`: Size of the entire endmember library.
+
+# Returns
+- A vector of integers representing the permuted endmember indices.
+"""
```

Comment:
I'm confused why `num_endmembers` must be a vector but only the first element is used. Seems like in other places it's a vector corresponding to number of endmembers for each class?

Reply:
This is fair...it's a vector because the treatment in some pathways (e.g. MESMA) does use the vector.  For SMA, only the first is used, which is a bit hidden.  Let's leave this as is, but convert the above comment to an issue, which we can think about how to tackle down the line.  Please resolve once the issue is posted.

Reply:
Issue #31 created